### DISCUSSION TITLE
[Q&A] Two diffrent fingerprints for different https adress.

### DISCUSSION THREAD

**Discussion by smurdza** [glpi-project/glpi-agent#393] [Q&A]
### Documentation & bug reporting acknowledgment

Yes, I read it

### Describe your problem

Hi,

I have a question.
Is it possible configure two different fingerprint for dwo different destination glpi server adres for htttps.

I have two servers glpi with they have two different domain.

How, Can I configured two different fingerprins ?

example case :

serwer =  https://glpi.asdf.pl,  https://glpi.qwert.pl
ssl-fingerprint = 'dkjfjfddkklss' , 'fkoewsnsis'   ????

**Comment by g-bougard** [comment]
Hello @smurdza 
this is a good question.
Actually, I think this is not possible, but looking at [IO::Socket::SSL SSL_fingerprint documentation](https://metacpan.org/pod/IO::Socket::SSL#SSL_fingerprint), I guess I can make a little change to support such kind of list.
Anyway, you also can use the same certificate on the 2 servers as ssl-fingerprint is made to bypass other normal checking like hostname.
Or you can also manage to have one of them signed by a public CA.

**Reply by smurdza** [reply]
Hi,
" I guess I can make a little change to support such kind of list."

**It will be nice... :)**

"you also can use the same certificate on the 2 servers as ssl-fingerprint is made to bypass other normal checking like hostname."
It is imposibble in my case.

"you can also manage to have one of them signed by a public CA."

I have 2 certyficates signed public CA.
Will I have to put CA certyfiace to ca-cert-dir ?

**Reply by g-bougard** [reply]
If your GLPI server SSL certificates are signed by public CA, you just don't need to use ssl-fingerprint or ca-cert-dir parameters.
You only need to use ssl-fingerprint, ca-certfile or ca-cert-dir when the public CA is not know by the agent. This can happen with public CAs which have been revoked or it is a too new one.

**Reply by g-bougard** [reply]
Hi @smurdza 

anyway, next nightly will support ssl-fingerprint as comma-separated list. See: https://github.com/glpi-project/glpi-agent/commit/283f86bff0ddd23290b2195e4d3bdf512b946db7

**Reply by smurdza** [reply]
@g-bougard 
Super, thank you :)

**Reply by smurdza** [reply]
@g-bougard 
I copy CA cert (PEM) to ca-cert-dir 
And I think that property doesn't work : 
[info] sending prolog request to server0
[error] [http client] internal response: 500 Can't connect to glpi.xyz.pl:443, IO::Socket::IP configuration failed error:140E0197:SSL routines:SSL_shutdown:shutdown while in init
[error] No answer from server at https://glpi.xyz.pl/marketplace/glpiinventory/

**Reply by g-bougard** [reply]
Hi @smurdza 
be careful about `ca-cert-dir` folder, just copying PEM file is not sufficient, they must have a specific name in that folder. Check the `CA_CERT_DIR=pathname` section in that documentation: https://glpi-agent.readthedocs.io/en/latest/installation/windows-command-line.html#command-line-parameters

**Reply by smurdza** [reply]
Ok, Its works. :+1:
